### PR TITLE
cli: Prepare login command for osemgrep invocation as default

### PR DESCRIPTION
**Pull request by zzeleznick**
## Description
This PR addresses some regression introduced with the login command and includes additional support for logging.

### Adds
- Common config to `login` and `logout` subcommands to make `--experimental` flag not error out
- Debugging and trace information in create login token 

### Fixes
- Login handler not animating the loading spinner
- Login command appearing to hang due to aggressive sleep times
- `get_deployment_from_token` vs `get_scan_config_from_token`

## Test Plan
- [x] Verified login now works
- [x] Verified correct headers are sent using nc

PR checklist:

- [x] Purpose of the code is [evident to future readers](https://semgrep.dev/docs/contributing/contributing-code/#explaining-code)
- [x] Tests included or PR comment includes a reproducible test plan
- [x] Documentation is up-to-date
- [x] A changelog entry was [added to changelog.d](https://semgrep.dev/docs/contributing/contributing-code/#adding-a-changelog-entry) for any user-facing change
- [x] Change has no security implications (otherwise, ping security team)

If you're unsure about any of this, please see:

- [Contribution guidelines](https://semgrep.dev/docs/contributing/contributing-code)!
- [One of the more specific guides located here](https://semgrep.dev/docs/contributing/contributing/)

### CODE REVIEW

**Review by ajbt200128 — APPROVED** [review]

File: PRIVACY.md [hunk 1]
```diff
@@ -193,21 +193,31 @@ Semgrep, Inc will:
 
 > `anonymous_user_id: "5f52484c-3f82-4779-9353-b29bbd3193b6"`
 
-This ID is stored in the `~/.semgrep/settings.yml` file.
-If the ID disappears, the next run will generate a new one randomly.
-The Semgrep team uses this to answer the following questions:
+We will generate a new random UUID and save the ID locally to a `~/.semgrep/settings.yml` file
```
Comment (pabloest) [inline PRIVACY.md]:
```suggestion
Semgrep generates a new random UUID and saves the ID locally to a `~/.semgrep/settings.yml` file
```

File: PRIVACY.md [hunk 2]
```diff
@@ -193,21 +193,31 @@ Semgrep, Inc will:
...
+when the ID is not present (and upon log out) to help improve the Semgrep product while commiting to promote online privacy and anonymity.
+
+The Semgrep team uses this ID to help answer the following questions:
 
 - > How many people use a given rule/ruleset/snippet?
 
-  This helps our security researchers assess their performance,
+  This enables our security researchers to assess their performance,
```
Comment (pabloest) [inline PRIVACY.md]:
```suggestion
  This enables the Semgrep Security Research Team to assess their performance,
```

File: PRIVACY.md [hunk 3]
```diff
@@ -193,21 +193,31 @@ Semgrep, Inc will:
...
   and we're planning to make these numbers public for all rule authors in the community.
 
-- > What percentage of users log in?"
+- > What percentage of users log in?
+
+  We use this to evaluate our success as we build new authenticated features for the Semgrep App.
```
Comment (pabloest) [inline PRIVACY.md]:
```suggestion
  We use this to evaluate our success as we build new authenticated features for the Semgrep Cloud Platform.
```

File: PRIVACY.md [hunk 4]
```diff
@@ -193,21 +193,31 @@ Semgrep, Inc will:
...
+By using privacy by design principles, we intend to limit the ability for this ID to be used to track users across the web.
 
-This ID will only ever be sent with Semgrep's metrics collection endpoint,
-meaning it cannot be used to track users across the web.
+1. **Randomization:** This ID is generated using a random algorithm (UUID V4) and does not contain any personally identifiable information (PII). Even with access to this ID, third-parties will not be able to associate this ID with a specific person.
+2. **Rotation:** This ID can be regularly reset, changed, or rotated, which limits entities to observe individual usage over an extended period of time.
+3. **Isolation:** This ID is only used for a specific purpose of analytics within a limited context of Semgrep and its product offerings.
+4. **Pseudonymization:** For logged in users who have accepted our terms of service and privacy policy, we will sanitize and transform (using a deterministic cryptographically secure hash) the PII used in generating a stable ID before sending data to our Semgrep, Inc backend and associated metrics collection endpoints.
```
Comment (pabloest) [inline PRIVACY.md]:
```suggestion
4. **Pseudonymization:** For logged in users who have accepted Semgrep’s terms of service and privacy policy, we will sanitize and transform (using a deterministic cryptographically secure hash) the PII used in generating a stable ID before sending data to the Semgrep, Inc backend and associated metrics collection endpoints.
```

**Review by pabloest — COMMENTED** [review]
copy edit suggestions

File: PRIVACY.md [hunk 5]
```diff
@@ -193,21 +193,23 @@ Semgrep, Inc will:
...
-The Semgrep team uses this to answer the following questions:
+Semgrep generates a new random UUID and saves the ID locally to a `~/.semgrep/settings.yml` file
+when the ID is not present to help improve Semgrep products.
+
+The Semgrep team uses this ID to help answer the following questions:
 
 - > How many people use a given rule/ruleset/snippet?
 
-  This helps our security researchers assess their performance,
+  This enables the Semgrep Security Research Team to assess their performance,
```
Comment (dlukeomalley) [inline PRIVACY.md]:
```suggestion
  This enables the Semgrep team to assess their performance,
```

**Review by dlukeomalley — APPROVED** [review]
Approving on the privacy clarifications. Defer to engineering on the implementation! 🎉

**Pull request merged**